src/hooks/prisma.ts

```diff
@@ -0,0 +1,137 @@
+import assert from "node:assert";
+
+import { ESTree } from "meriyah";
+import type prisma from "@prisma/client";
+
+import type AppMap from "../AppMap";
+import { getTime } from "../util/getTime";
+import { fixReturnEventIfPromiseResult, recording } from "../recorder";
+import { FunctionInfo } from "../registry";
+
+export default function prismaHook(mod: typeof prisma) {
+  // Imported PrismaClient type does not have _request method in type definition.
+  // But we have it in runtime.
+  assert("PrismaClient" in mod);
+  assert(mod.PrismaClient != null);
+  const PC = mod.PrismaClient as { prototype: unknown };
+  const proto = PC.prototype;
+  assert(proto != null && typeof proto === "object");
+  assert("_request" in proto);
+  proto._request = createProxy(proto._request as (...args: unknown[]) => unknown);
+  return mod;
+}
+
+prismaHook.applicable = function (id: string) {
+  return ["@prisma/client"].includes(id);
```

Comment:
Can we configure this? We have internal services which share a pre-configured, pre-built Prisma client, shipped as its own module. Because of this, we don't ever import `@prisma/client` directly.

Reply:
It's worth noting that I've tried this PR with our own service after modifying this line and it does work great.

Reply:
I'll make the module name (id) a configuration parameter, if I understand correctly.